src/librustc_resolve/macros.rs

```diff
@@ -449,7 +449,16 @@ impl<'a, 'cl> Resolver<'a, 'cl> {
             }
         };
 
-        let path = attr.as_ref().unwrap().path.clone();
+        let path = {
+            let attr = attr.as_ref().unwrap();
+
+            // A hack to make #[test] not shadowable. This is in line with legacy behavior.
```

Comment:
This check seems a bit suspect to me in the sense that we're in general going towards a world where it's fine to shadow built-in attributes (for the most part). Instead of this though I'd prefer to make it an error to define or import a macro called `test` rather than forcing resolution always go to the macro prelude.

Could you elaborate a bit though on how this came up?

Reply:
Sure thing, there's this phenomenon where people have defined macro_rules macros called `test!` that they use as some part of generating tests. For example:
https://github.com/rust-lang/rust/blob/c24f27c0452a9aaecfb70d225d846df2e517385f/src/libsyntax_pos/analyze_source_file.rs#L288-L297


We see this in many places even just within the compiler but a cursory github search also reveals there are many.

No one would be happier than me to remove this hack, but I figured it best to start conservative and remove later if we can.